packages/eyeglass/src/modules/EyeglassModule.ts

```diff
@@ -1,44 +1,201 @@
 import * as packageUtils from "../util/package";
-import * as merge from "lodash.merge";
-import * as includes from "lodash.includes";
+import merge = require("lodash.merge");
+import includes = require("lodash.includes");
 import * as path from "path";
 import * as fs from "fs";
+import { IEyeglass } from "../IEyeglass";
+import { FunctionDeclarations, SassImplementation } from "../util/SassImplementation";
+import { PackageJson } from "package-json";
+import AssetsCollection from "../assets/AssetsCollection";
+import { Dict, isPresent } from "../util/typescriptUtils";
 
-var rInvalidName = /\.(?:sass|s?css)$/;
-var EYEGLASS_KEYWORD = "eyeglass-module";
+const rInvalidName = /\.(?:sass|s?css)$/;
+const EYEGLASS_KEYWORD: "eyeglass-module" = "eyeglass-module";
 
-export default class EyeglassModule {
-  eyeglass: any;
+export interface DiscoverOptions {
+  isRoot: boolean;
+  dir: string;
+  pkg?: packageUtils.Package
+}
+
+export interface EyeglassModuleExports {
+  name?: string;
+  functions?: FunctionDeclarations;
+  assets?: AssetsCollection;
+  sassDir?: string;
+  eyeglass?: {
+    needs?: string;
+  };
+}
+
+interface EyeglassModuleOptionsFromPackageJSON {
+  name?: string;
+  exports?: string | false;
+  sassDir?: string;
+  needs?: string;
+}
+
+interface PackageEyeglassOption {
+  eyeglass?: string | EyeglassModuleOptionsFromPackageJSON;
+}
+type PackageJsonWithEyeglassOptions = PackageJson & PackageEyeglassOption;
+
+type EyeglassOptionInPackageJSON = PackageJsonWithEyeglassOptions["eyeglass"];
+
+export type EyeglassModuleMain =
+  (eyeglass: IEyeglass, sass: SassImplementation) => EyeglassModuleExports;
+
+export type ModuleSpecifier = ModuleReference | ManualModuleOptions;
+
+export interface ModuleReference {
+  path: string;
+  /**
+   * XXX I don't think dependencies are ever actually
+   * passed along with a path reference but the code allows it.
+   */
+  dependencies?: Array<EyeglassModule>;
+  isEyeglassModule?: boolean;
+}
+export interface ManualModuleOptions {
+  /**
+   * The name of the module.
+   */
+  name?: string;
+  /**
+   * The function that would normally be exported from the
+   * eyeglass exports file.
+   */
+  main?: EyeglassModuleMain | null;
+  /**
+   * If a main function is provided this path to the 
+   * filename where it is defined should be provided
+   * for better error messages in some situations.
+   */
+  mainPath?: string | undefined;
```

Comment:
optional _and_ `| undefined` seems redundant

Reply:
Fixed in 70977805c91e1bce0957e8ff157f3fecaceda3a7